src/core/reference/src/op/concat.cpp

```diff
@@ -17,55 +17,64 @@ std::vector<size_t> calculate_shape_sizes(const std::vector<Shape>& in_shapes) {
     });
     return sizes;
 }
-}  // namespace
-
-void concat(const std::vector<const char*>& args,
-            char* out,
-            const std::vector<Shape>& in_shapes,
-            const Shape& out_shape,
-            int64_t concatenation_axis,
-            size_t elem_size) {
-    size_t steps = 1;
-    for (int i = 0; i < concatenation_axis; ++i) {
-        steps *= out_shape[i];
-    }
-
-    const auto& shape_sizes = calculate_shape_sizes(in_shapes);
-
-    size_t out_offset = 0;
-    for (size_t step = 0; step < steps; ++step) {
-        for (size_t in_index = 0; in_index < args.size(); ++in_index) {
-            const size_t size = shape_sizes[in_index] / steps;
-            const size_t in_offset = step * size;
+inline void copy_single_input_elements(const char* arg,
+                                       char* out,
+                                       size_t in_offset,
+                                       size_t out_offset,
+                                       size_t num_of_elements,
+                                       size_t elem_size) {
+    std::memcpy(&out[out_offset * elem_size], &arg[in_offset * elem_size], num_of_elements * elem_size);
```

Comment:
What about direct pointer arithmetic?  
```suggestion
    std::memcpy(out + out_offset * elem_size, arg + in_offset * elem_size, num_of_elements * elem_size);
```

Reply:
Done